src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -35,13 +35,25 @@
 _TYPES_DICT[types.FunctionType] = "FunctionType"
 
 
+def _get_types_type(obj: Any) -> str | None:
+    try:
+        return _TYPES_DICT.get(obj)
+    except TypeError:
+        # e.g. exception: unhashable type
+        return None
```

Comment:
I think the following would be a bit more explicit:
```suggestion
from collections.abc import Hashable

def _get_types_type(obj: Any) -> str | None:
    if not isinstance(obj, Hashable):
        return None
    return _TYPES_DICT.get(obj)
```

Reply:
tbh. I even thought about
```python
def _get_types_type(obj: Any) -> str | None:
    try:
        return _TYPES_DICT.get(obj)
    except:
        # e.g. exception: unhashable type
        return None
```
because does it really matter why the lookup fails? 
If the value can't be looked up it sure is not in the dictionary.

While `Hashable` might work for ParamSpec it's [not guaranteed](https://docs.python.org/3/library/collections.abc.html#id17) to work every time.
I think catching the (broad) exception would be the best solution programmatically but if you want me to use `Hashable` I'll change it accordingly

Reply:
I usually prefer starting out by being more specific since that way we find out if our assumptions are wrong. But certainly if someone does something like:
```py
class BadClass:
    def __hash__(self):
        raise TypeError("Oops!")
```
then the class would count as hashable but attempts to hash it would always raise an error. Probably any of these things are fine.